internal/provider/serverless_cluster_resource.go

```diff
@@ -0,0 +1,813 @@
+	if err != nil {
+		resp.Diagnostics.AddError("Create Error", fmt.Sprintf("Unable to call CreateCluster, got error: %s", err))
+		return
+	}
+
+	clusterId := *cluster.ClusterId
+	data.ClusterId = types.StringValue(clusterId)
+	tflog.Info(ctx, "wait serverless cluster ready")
+	cluster, err = WaitServerlessClusterReady(ctx, clusterServerlessCreateTimeout, clusterServerlessCreateInterval, clusterId, r.provider.ServerlessClient)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Cluster creation failed",
+			fmt.Sprintf("Cluster is not ready, get error: %s", err),
+		)
+		return
+	}
+	cluster, err = r.provider.ServerlessClient.GetCluster(ctx, *cluster.ClusterId, clusterV1beta1.SERVERLESSSERVICEGETCLUSTERVIEWPARAMETER_FULL)
+	if err != nil {
+		tflog.Error(ctx, fmt.Sprintf("Unable to call GetCluster, error: %s", err))
+		return
+	}
+	err = refreshServerlessClusterResourceData(ctx, cluster, &data)
+	if err != nil {
+		resp.Diagnostics.AddError("Refresh Error", fmt.Sprintf("Unable to refresh serverless cluster resource data, got error: %s", err))
+		return
+	}
+
+	// save into the Terraform state.
+	diags = resp.State.Set(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r serverlessClusterResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	// get data from state
+	var data serverlessClusterResourceData
+	diags := req.State.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	tflog.Debug(ctx, fmt.Sprintf("read serverless_cluster_resource clusterid: %s", data.ClusterId.ValueString()))
+
+	// call read api
+	tflog.Trace(ctx, "read serverless_cluster_resource")
+	cluster, err := r.provider.ServerlessClient.GetCluster(ctx, data.ClusterId.ValueString(), clusterV1beta1.SERVERLESSSERVICEGETCLUSTERVIEWPARAMETER_FULL)
+	if err != nil {
+		tflog.Error(ctx, fmt.Sprintf("Unable to call GetCluster, error: %s", err))
+		return
+	}
+	err = refreshServerlessClusterResourceData(ctx, cluster, &data)
+	if err != nil {
+		resp.Diagnostics.AddError("Refresh Error", fmt.Sprintf("Unable to refresh serverless cluster resource data, got error: %s", err))
+		return
+	}
+
+	// save into the Terraform state.
+	diags = resp.State.Set(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r serverlessClusterResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var clusterId string
+
+	resp.Diagnostics.Append(req.State.GetAttribute(ctx, path.Root("cluster_id"), &clusterId)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Trace(ctx, "delete serverless_cluster_resource")
+	_, err := r.provider.ServerlessClient.DeleteCluster(ctx, clusterId)
+	if err != nil {
+		resp.Diagnostics.AddError("Delete Error", fmt.Sprintf("Unable to call DeleteCluster, got error: %s", err))
+		return
+	}
+}
+
+func (r serverlessClusterResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
I don't get Update. I know our Open API limit one field modification a time. And the code does this. If the user updates several fields a time what happens now?